### DISCUSSION TITLE
Is it possible for a new contract to interact with another old contract function which has been deprecated

### DISCUSSION THREAD

**Discussion by drained69** [Cyfrin/security-and-auditing-full-course-s23#18] [Ideas]
Currently on Patrick's security course on selfDestruct, and when trying to test the section myself from solidity by example, I noticed that the selfDestruct has been deprecated. Would love to hear people's opinions on the best steps to take when they notice that their old contracts have some deprecated function, and also whether it is possible to interact with old contracts functions that are deprecated.

**Comment by cromewar** [comment]
@Chinwuba22 
You would have to stick to the old version to interact with that specific deprecated function. and that applies both for solidity and externals dependencies such as openzeppelin and chainlink.

**Reply by drained69** [reply]
Alright thanks sir
